Source/WebCore/inspector/InspectorOverlay.cpp

```diff
@@ -1548,11 +1548,27 @@ std::optional<InspectorOverlay::Highlight::GridHighlightOverlay> InspectorOverla
     auto rowPositions = renderGrid.rowPositions();
     if (!columnPositions.size() || !rowPositions.size())
         return { };
-    
+
+    LayoutUnit masonryContentSize = renderGrid.masonryContentSize();
+
+    // There are no actual rows or columns in the masonry axis of a masonry layout.
+    // But we can borrow the concept to draw the two lines at the start and end of the masonry axis.
+    if (renderGrid.areMasonryRows()) {
+        auto firstRowPosition = rowPositions[0];
+        auto lastRowPosition = masonryContentSize;
```

Comment:
Will the first row/column position always be 0? Or should `lastRowPosition` actually equal `firstRowPosition + masonryContentSize`?

Reply:
Good catch! 👍 

No, the first row (or column) position isn't always zero because it includes the border, padding and offsets for the grid container itself:
https://github.com/WebKit/WebKit/blob/f6ce8ab19c996c796d7ae5c6fa23ab6a27e64678/Source/WebCore/rendering/RenderGrid.cpp#L1492

Fixed in the latest patch.

**Screenshot before**
<img width="1392" alt="Screenshot 2023-11-29 at 21 05 55" src="https://github.com/WebKit/WebKit/assets/105809798/74a8034c-8d25-4653-af82-c27e7d5b5794">


**Screenshot after**
<img width="1392" alt="Screenshot 2023-11-29 at 21 00 55" src="https://github.com/WebKit/WebKit/assets/105809798/40332104-9087-4aa6-93b1-c00d5b3454c8">

(the gap between the last item and the bottom line is the default margin for a `<p>` from the browser styles)